Documentation/network/servicemesh/envoy-traffic-shifting.rst

```diff
@@ -0,0 +1,154 @@
+    helloworld-v1-5845f97d6b-gkdtk   1/1     Running   0          2m23s   10.0.0.241   cilium-control-plane   <none>           <none>            app=helloworld,pod-template-hash=5845f97d6b,version=v1
+    helloworld-v2-7d55d87964-ns9kh   1/1     Running   0          2m23s   10.0.0.251   cilium-control-plane   <none>           <none>            app=helloworld,pod-template-hash=7d55d87964,version=v2
+
+    $ kubectl get svc --show-labels
+    NAME                              READY   STATUS    RESTARTS   AGE   IP           NODE                       NOMINATED NODE   READINESS GATES   LABELS
+    helloworld      ClusterIP   10.96.194.77   <none>        5000/TCP   8m27s   app=helloworld,service=helloworld
+
+Apply weight-based routing
+==========================
+
+Make an environment variable with the Pod name for client:
+
+.. code-block:: shell-session
+
+    $ export CLIENT=$(kubectl get pods -l name=client -o jsonpath='{.items[0].metadata.name}')
+
+Try making several requests to the helloworld Service.
+
+.. code-block:: shell-session
+
+    $ for i in {1..10}; do  kubectl exec -it $CLIENT -- curl  helloworld:5000/hello; done
+
+The test results are as follows::
+
+    Hello version: v2, instance: helloworld-v2-7d55d87964-ns9kh
+    Hello version: v2, instance: helloworld-v2-7d55d87964-ns9kh
+    Hello version: v2, instance: helloworld-v2-7d55d87964-ns9kh
+    Hello version: v1, instance: helloworld-v1-5845f97d6b-gkdtk
+    Hello version: v1, instance: helloworld-v1-5845f97d6b-gkdtk
+    Hello version: v1, instance: helloworld-v1-5845f97d6b-gkdtk
+    Hello version: v2, instance: helloworld-v2-7d55d87964-ns9kh
+    Hello version: v1, instance: helloworld-v1-5845f97d6b-gkdtk
+    Hello version: v2, instance: helloworld-v2-7d55d87964-ns9kh
+    Hello version: v1, instance: helloworld-v1-5845f97d6b-gkdtk
+
+The test results were as expected. Of the requests sent to the helloworld service,
+50% of them were sent to backend ``helloworld-v1`` and 50% of them were sent to
+the backend ``helloworld-v2``.
+
+Double-check whether your backend service is up and healthy.
+The Envoy Discovery Service (EDS) has a name that follows the convention ``<namespace>/<service-name>:<port>``.
+
+So, you need to define a different Kubernetes Service for each version.
+
+.. parsed-literal::
+
+    $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/envoy/helloworld-service-v1-v2.yaml
+
+
+Apply the ``envoy-helloworld-v1-90-v2-10.yaml`` file, which defines a ``CiliumEnvoyConfig``. 
+``CiliumEnvoyConfig`` can be used to load balance traffic destined to one Service to a
+group of backend Services. To load balance traffic to the helloworld Service, first create
+individual Services for each backend Deployment.
+
+.. parsed-literal::
+
+    $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/envoy/envoy-helloworld-v1-90-v2-10.yaml
+
```

Comment:
Do you think we could reorder this section a bit? Also, I'm not sure that mentioning the Envoy Discovery Service naming convention is too useful here, as the user can pick up on this by reading through the `CiliumEnvoyConfig` that you provided.

```suggestion
``CiliumEnvoyConfig`` can be used to load balance traffic destined to one Service to a
group of backend Services. To load balance traffic to the helloworld Service, first create
individual Services for each backend Deployment.

.. parsed-literal::

    $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/envoy/helloworld-service-v1-v2.yaml
 
Apply the ``envoy-helloworld-v1-90-v2-10.yaml`` file, which defines a ``CiliumEnvoyConfig`` to send 90% of traffic to the helloworld-v1 Service backend and 10% of traffic to the hellworld-v2 Service backend:

.. parsed-literal::

    $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/envoy/envoy-helloworld-v1-90-v2-10.yaml
```

Reply:
ok I updated it.